plutarch-ledger-api/src/Plutarch/LedgerApi/Interval.hs

```diff
@@ -273,10 +296,112 @@ pmember ::
     )
 pmember = phoistAcyclic $ plam $ \x i -> pcontains # i # (psingleton # x)
 
-{- | @'pcontains' # x # y@ is true if the interval @y@ is entirely contained in @a@.
+{- | Check if a 'PInterval' is empty.
 
-@since 2.1.1
+@since WIP
 -}
+pisEmpty ::
+  forall (a :: S -> Type) (s :: S).
+  (PIsData a, PEnumerable a) =>
+  Term s (PInterval a :--> PBool)
+pisEmpty = phoistAcyclic $ plam $ \i -> unTermCont $ do
+  unpacked <- tcont $ pletFields @'["from", "to"] i
+  let lowerBound = getField @"from" unpacked
+  let upperBound = getField @"to" unpacked
+  let inclusiveLowerBound = pinclusiveLowerBound # lowerBound
+  let inclusiveUpperBound = pinclusiveUpperBound # upperBound
+  pure $ inclusiveLowerBound #> inclusiveUpperBound
+
+{- | Turn a 'PLowerBound' into a single inclusive bounding value.
+
+@since WIP
+-}
+pinclusiveLowerBound ::
+  forall (a :: S -> Type) (s :: S).
+  (PIsData a, PCountable a) =>
+  Term s (PLowerBound a :--> PExtended a)
+pinclusiveLowerBound = phoistAcyclic $ plam $ \lb -> unTermCont $ do
+  unpacked <- tcont $ pletFields @'["_0", "_1"] lb
+  let extended = getField @"_0" unpacked
```

Comment:
as `extended` is used twice, once each on `pif` this should technically use `plet` but I think it's fine here

Reply:
Easy to change.

Reply:
On second look, this isn't the best idea:

`  ./Plutarch/Test/Golden.hs:256:3: 
  2) Plutarch.Extra.Interval.extra.intervalutils.fixtures.golden bench
       expected: constants.always {"exBudgetCPU":16100,"exBudgetMemory":200,"scriptSizeBytes":34}
                 contains.in interval {"exBudgetCPU":28461443,"exBudgetMemory":97463,"scriptSizeBytes":553}
                 contains.out interval {"exBudgetCPU":22778323,"exBudgetMemory":79756,"scriptSizeBytes":552}
                 contains.always {"exBudgetCPU":22193602,"exBudgetMemory":75658,"scriptSizeBytes":561}
                 member.[b,c], a < b {"exBudgetCPU":22459810,"exBudgetMemory":78292,"scriptSizeBytes":562}
                 member.[b,c], a = b {"exBudgetCPU":28094930,"exBudgetMemory":95699,"scriptSizeBytes":556}
                 member.[b,c], a > b, a < c {"exBudgetCPU":28142930,"exBudgetMemory":95999,"scriptSizeBytes":562}
                 member.[b,c], a = c {"exBudgetCPU":28142930,"exBudgetMemory":95999,"scriptSizeBytes":559}
                 member.[b,c], a > c {"exBudgetCPU":28142930,"exBudgetMemory":95999,"scriptSizeBytes":562}
                 hull.hull 3 5 contains 3 5 {"exBudgetCPU":45167369,"exBudgetMemory":156699,"scriptSizeBytes":999}
                 hull.2 not member of hull 3 5 {"exBudgetCPU":39309736,"exBudgetMemory":138428,"scriptSizeBytes":982}
                 hull.6 not member of hull 3 5 {"exBudgetCPU":39309736,"exBudgetMemory":138428,"scriptSizeBytes":982}
                 intersection.intesection [2,4] [3,5] contains [3,4] {"exBudgetCPU":45708395,"exBudgetMemory":159027,"scriptSizeBytes":999}
                 intersection.intesection [3,5] [2,4] contains [3,4] {"exBudgetCPU":51016829,"exBudgetMemory":177081,"scriptSizeBytes":999}
        but got: constants.always {"exBudgetCPU":16100,"exBudgetMemory":200,"scriptSizeBytes":34}
                 contains.in interval {"exBudgetCPU":28653443,"exBudgetMemory":98663,"scriptSizeBytes":552}
                 contains.out interval {"exBudgetCPU":22970323,"exBudgetMemory":80956,"scriptSizeBytes":551}
                 contains.always {"exBudgetCPU":22385602,"exBudgetMemory":76858,"scriptSizeBytes":560}
                 member.[b,c], a < b {"exBudgetCPU":22651810,"exBudgetMemory":79492,"scriptSizeBytes":561}
                 member.[b,c], a = b {"exBudgetCPU":28286930,"exBudgetMemory":96899,"scriptSizeBytes":555}
                 member.[b,c], a > b, a < c {"exBudgetCPU":28334930,"exBudgetMemory":97199,"scriptSizeBytes":561}
                 member.[b,c], a = c {"exBudgetCPU":28334930,"exBudgetMemory":97199,"scriptSizeBytes":557}
                 member.[b,c], a > c {"exBudgetCPU":28334930,"exBudgetMemory":97199,"scriptSizeBytes":561}
                 hull.hull 3 5 contains 3 5 {"exBudgetCPU":45359369,"exBudgetMemory":157899,"scriptSizeBytes":998}
                 hull.2 not member of hull 3 5 {"exBudgetCPU":39501736,"exBudgetMemory":139628,"scriptSizeBytes":980}
                 hull.6 not member of hull 3 5 {"exBudgetCPU":39501736,"exBudgetMemory":139628,"scriptSizeBytes":980}
                 intersection.intesection [2,4] [3,5] contains [3,4] {"exBudgetCPU":45900395,"exBudgetMemory":160227,"scriptSizeBytes":997}
                 intersection.intesection [3,5] [2,4] contains [3,4] {"exBudgetCPU":51208829,"exBudgetMemory":178281,"scriptSizeBytes":998}
`

Gonna leave it as-is.